framework/db/Connection.php

```diff
@@ -1004,7 +1009,7 @@ function ($matches) {
     public function getDriverName()
     {
         if ($this->_driverName === null) {
-            if (($pos = strpos($this->dsn, ':')) !== false) {
+            if (($pos = strpos(isset($this->dsn) ? $this->dsn : '', ':')) !== false) {
```

Comment:
```suggestion
            if (($pos = strpos((string)$this->dsn), ':')) !== false) {
```

Reply:
@samdark there's a syntax error in here

Reply:
yes, it's one redundant close parenthesis after `$this->dsn`